db/migrate/20250330181923_add_user_ref_to_recipes.rb

```diff
@@ -0,0 +1,6 @@
+class AddUserRefToRecipes < ActiveRecord::Migration[7.2]
+  Recipe.delete_all
```

Comment:
Is there a reason why this isn't inside the method? Does it work like this?

Reply:
Interesting. So I think the way this is, it will delete all the recipes when rolling back as well as migrating.  If you wanted to only delete the recipes on migrate, you'd want to do something like:

```
def up
  Recipe.delete_all
  add_reference :recipes, :user, null: false, foreign_key: true
end

def down
  remove_reference :recipes, :user, null: false, foreign_key: true
end
```

Reply:
So do I need to do a rollback on this specific migration and re migrate to fix this?

Reply:
That would be the proper way to do it, yes.

You could also checkout the main branch, reset the db, then switch back to this branch, update the migration script and then apply